lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -317,7 +318,6 @@ def add_payment(post, payment)
         if payment.is_a?(String)
           _, _, recurring_detail_reference = payment.split('#')
           post[:selectedRecurringDetailReference] = recurring_detail_reference
-          add_recurring_contract(post, options)
```

Comment:
Why is this not needed anymore? I got the impression that we needed to add this method to `authorize`, but I'm not sure why it's been removed here.

Reply:
`add_payment` only gets called in `authorize` and `store`; since `add_recurring_contract` is now included directly in those methods, we didn't need it here (and it seemed to me like `add_recurring_contract` made sense being its own entity, rather than part of `add_payment`.)

Reply:
But I probably need to handle it better, one moment

Reply:
Okay, two moments 😅